s3fs/mapwrap.py

```diff
@@ -0,0 +1,73 @@
+
+from collections import MutableMapping
+import os
+
+class MapWrap(MutableMapping):
+    """Wrap an S3FileSystem as a mutable wrapping.
+
+    The keys of the mapping become files under the given root, and the
+    values (which must be bytes) the contents of those files.
+
+    Usage
+    -----
+    >>> mw = MapWrapping(s3, 'mybucket/mapstore/') # doctest: +SKIP
+    >>> mw['loc1'] = b'Hello World' # doctest: +SKIP
+    >>> list(mw.keys()) # doctest: +SKIP
+    ['loc1']
+    >>> mw['loc1'] # doctest: +SKIP
+    b'Hello World'
+    """
+
+    def __init__(self, s3, root, check=False):
+        """A mutable mapping at the given S3 location
+
+        Parameters
+        ----------
+        s3 : S3FileSystem
+        root : string
+            prefix for all the files (perhaps justa  bucket name
+        check : bool (=True)
+            performs a touch at the location, to check writeability.
+        """
+        self.s3 = s3
+        self.root = root
+        if check:
+            s3.touch(root+'/a')
+            s3.rm(root+'/a')
+
+    def clear(self):
+        """Remove all keys below root - empties out mapping
+        """
+        self.s3.rm(self.root, recursive=True)
+
+    def __getitem__(self, key):
```

Comment:
Or maybe a general function 

```python
key = normalize_key(key)

def normalize_key(key):
    if isinstance(key, tuple):
        return '/'.join(map(normalize_key, key))
    elif isinstance(key, str):
        return key
    else:
        return key
```

This way we can change around the key naming policy easily in the future if necessary.